src/TMxpMusicTagHandler.cpp

```diff
@@ -29,7 +29,9 @@ TMxpTagHandlerResult TMxpMusicTagHandler::handleStartTag(TMxpContext& ctx, TMxpC
 
     mediaData.setMediaProtocol(TMediaData::MediaProtocolMSP);
     mediaData.setMediaType(TMediaData::MediaTypeMusic);
-    mediaData.setMediaFileName(tag->getAttributeValue("FName"));
+    if (tag->hasAttribute("FName")) {
```

Comment:
Every MXP MSP tag should have an FName (which is either `Off` or a file name). I recall elsewhere in the code where there was some validation. I wonder if adding this here may do more harm than good - or perhaps have no effect due to validation elsewhere. 

If this change is appropriate here, it may also need added to TMxpSoundTagHandler::handleStartTag as well.

Reply:
This could be the actual bugfix...

Reply:
This was necessary in combination with the other bugfix during my development. If you're not sold feel free to replicate the scenario!

Reply:
I do not have the confidence/ability/any MUDS setup to frob with MXP stuff - but I feel sure that removing the explicit initialisations of those `QString` members on lines 86 and 91 to 94 will not cause the bug to return.